src/features/orderBy/server/orderByToPrisma.ts

```diff
@@ -0,0 +1,30 @@
+import { type OrderByState } from "@/src/features/orderBy/types";
+import { type ColumnDefinition } from "@/src/server/api/interfaces/tableDefinition";
+import { Prisma } from "@prisma/client";
+
+/**
+ * Convert orderBy to SQL ORDER BY clause
+ * @param orderBy orderBy state
+ * @param tableColumns column definitions, used to map orderBy column to internal name
+ * @returns Prisma.sql
+ */
+export function orderByToPrismaSql(
+  orderBy: OrderByState,
+  tableColumns: ColumnDefinition[],
+): Prisma.Sql {
+  if (!orderBy) {
+    return Prisma.sql([`ORDER BY t.timestamp DESC`]);
+  }
+  // Get column definition to map column to internal name, e.g. "t.id"
+  const col = tableColumns.find(
+    // TODO: Only use id instead of name.
+    // It's less error-prone & decouples data fetching from the human-readable UI labels
```

Comment:
We're right now using `.name` to match the table column, which seems very error-prone to me since that's the UI copy - API behaviour should never depend on the copy on the UI imo :)

I didn't want to overblow this PR by doing this migration here, but if you're happy to do it & don't see any risks with it, i can also do it here.